diff -ur MigrationTools-47/migrate_aliases.pl MigrationTools-47-etc/migrate_aliases.pl
--- MigrationTools-47/migrate_aliases.pl	2006-01-25 05:18:16.000000000 +0100
+++ MigrationTools-47-etc/migrate_aliases.pl	2007-11-05 17:17:05.880512377 +0100
@@ -36,7 +36,7 @@
 # thanks to Dave McPike
 #
 
-require 'migrate_common.ph';
+require '%CONFDIR%migrate_common.ph';
 
 $PROGRAM = "migrate_aliases.pl";
 $NAMINGCONTEXT = &getsuffix($PROGRAM);
diff -ur MigrationTools-47/migrate_all_offline.sh MigrationTools-47-etc/migrate_all_offline.sh
--- MigrationTools-47/migrate_all_offline.sh	2006-01-25 05:18:16.000000000 +0100
+++ MigrationTools-47-etc/migrate_all_offline.sh	2007-11-05 17:17:05.887178585 +0100
@@ -34,7 +34,7 @@
 #
 # Migrate all entities from flat files. 
 #
-# Make sure that you configure migrate_common.ph to suit
+# Make sure that you configure @@CONFDIR%migrate_common.ph to suit
 # your site's X.500 naming context and DNS mail domain;
 # the defaults may not be correct.
 #
diff -ur MigrationTools-47/migrate_all_online.sh MigrationTools-47-etc/migrate_all_online.sh
--- MigrationTools-47/migrate_all_online.sh	2006-01-25 05:18:16.000000000 +0100
+++ MigrationTools-47-etc/migrate_all_online.sh	2007-11-05 17:17:05.890511690 +0100
@@ -104,7 +104,7 @@
 fi
 
 if [ "X$LDAP_BASEDN" = "X" ]; then
-	defaultcontext=`$PERL -e 'require "migrate_common.ph"; print \$DEFAULT_BASE';`
+	defaultcontext=`$PERL -e 'require "%CONFDIR%migrate_common.ph"; print \$DEFAULT_BASE';`
 	question="Enter the X.500 naming context you wish to import into: [$defaultcontext]"
 	echo "$question " | tr -d '\012' > /dev/tty
 	read LDAP_BASEDN
diff -ur MigrationTools-47/migrate_automount.pl MigrationTools-47-etc/migrate_automount.pl
--- MigrationTools-47/migrate_automount.pl	2006-01-25 05:18:16.000000000 +0100
+++ MigrationTools-47-etc/migrate_automount.pl	2007-11-05 17:17:05.890511690 +0100
@@ -33,7 +33,7 @@
 # services migration tool
 #
 
-require 'migrate_common.ph';
+require '%CONFDIR%migrate_common.ph';
  
 $PROGRAM = "migrate_automount.pl"; 
 $NAMINGCONTEXT = &getsuffix($PROGRAM);
diff -ur MigrationTools-47/migrate_base.pl MigrationTools-47-etc/migrate_base.pl
--- MigrationTools-47/migrate_base.pl	2006-01-25 05:18:16.000000000 +0100
+++ MigrationTools-47-etc/migrate_base.pl	2007-11-05 17:17:05.890511690 +0100
@@ -36,7 +36,7 @@
 #
 #
 
-require 'migrate_common.ph';
+require '%CONFDIR%migrate_common.ph';
 
 $PROGRAM = "migrate_base.pl";
 $NAMINGCONTEXT = &getsuffix($PROGRAM);
diff -ur MigrationTools-47/migrate_fstab.pl MigrationTools-47-etc/migrate_fstab.pl
--- MigrationTools-47/migrate_fstab.pl	2006-01-25 05:18:16.000000000 +0100
+++ MigrationTools-47-etc/migrate_fstab.pl	2007-11-05 17:17:05.893844794 +0100
@@ -37,7 +37,7 @@
 # They are used by MacOS X Server, however.
 #
 
-require 'migrate_common.ph';
+require '%CONFDIR%migrate_common.ph';
 
 $PROGRAM = "migrate_fstab.pl";
 $NAMINGCONTEXT = &getsuffix($PROGRAM);
diff -ur MigrationTools-47/migrate_group.pl MigrationTools-47-etc/migrate_group.pl
--- MigrationTools-47/migrate_group.pl	2006-01-25 05:18:16.000000000 +0100
+++ MigrationTools-47-etc/migrate_group.pl	2007-11-05 17:17:05.893844794 +0100
@@ -36,7 +36,7 @@
 #
 #
 
-require 'migrate_common.ph';
+require '%CONFDIR%migrate_common.ph';
 
 $PROGRAM = "migrate_group.pl";
 $NAMINGCONTEXT = &getsuffix($PROGRAM);
diff -ur MigrationTools-47/migrate_hosts.pl MigrationTools-47-etc/migrate_hosts.pl
--- MigrationTools-47/migrate_hosts.pl	2006-01-25 05:18:16.000000000 +0100
+++ MigrationTools-47-etc/migrate_hosts.pl	2007-11-05 17:17:05.893844794 +0100
@@ -36,7 +36,7 @@
 #
 #
 
-require 'migrate_common.ph';
+require '%CONFDIR%migrate_common.ph';
 
 $PROGRAM = "migrate_hosts.pl";
 $NAMINGCONTEXT = &getsuffix($PROGRAM);
diff -ur MigrationTools-47/migrate_netgroup_byhost.pl MigrationTools-47-etc/migrate_netgroup_byhost.pl
--- MigrationTools-47/migrate_netgroup_byhost.pl	2006-01-25 05:18:16.000000000 +0100
+++ MigrationTools-47-etc/migrate_netgroup_byhost.pl	2007-11-05 17:17:05.893844794 +0100
@@ -36,7 +36,7 @@
 #
 #
 
-require 'migrate_common.ph';
+require '%CONFDIR%migrate_common.ph';
 
 $PROGRAM = "migrate_netgroup_byhost.pl";
 $NAMINGCONTEXT = &getsuffix($PROGRAM);
diff -ur MigrationTools-47/migrate_netgroup_byuser.pl MigrationTools-47-etc/migrate_netgroup_byuser.pl
--- MigrationTools-47/migrate_netgroup_byuser.pl	2006-01-25 05:18:16.000000000 +0100
+++ MigrationTools-47-etc/migrate_netgroup_byuser.pl	2007-11-05 17:17:05.893844794 +0100
@@ -36,7 +36,7 @@
 #
 #
 
-require 'migrate_common.ph';
+require '%CONFDIR%migrate_common.ph';
 
 $PROGRAM = "migrate_netgroup_byuser.pl";
 $NAMINGCONTEXT = &getsuffix($PROGRAM);
diff -ur MigrationTools-47/migrate_netgroup.pl MigrationTools-47-etc/migrate_netgroup.pl
--- MigrationTools-47/migrate_netgroup.pl	2006-01-25 05:18:16.000000000 +0100
+++ MigrationTools-47-etc/migrate_netgroup.pl	2007-11-05 17:17:05.897177898 +0100
@@ -36,7 +36,7 @@
 # line continuation support by Bob Apthorpe
 #
 
-require 'migrate_common.ph';
+require '%CONFDIR%migrate_common.ph';
 
 $PROGRAM = "migrate_netgroup.pl";
 $NAMINGCONTEXT = &getsuffix($PROGRAM);
diff -ur MigrationTools-47/migrate_networks.pl MigrationTools-47-etc/migrate_networks.pl
--- MigrationTools-47/migrate_networks.pl	2006-01-25 05:18:16.000000000 +0100
+++ MigrationTools-47-etc/migrate_networks.pl	2007-11-05 17:17:05.897177898 +0100
@@ -36,7 +36,7 @@
 #
 #
 
-require 'migrate_common.ph';
+require '%CONFDIR%migrate_common.ph';
 
 $PROGRAM = "migrate_networks.pl";
 $NAMINGCONTEXT = &getsuffix($PROGRAM);
diff -ur MigrationTools-47/migrate_passwd.pl MigrationTools-47-etc/migrate_passwd.pl
--- MigrationTools-47/migrate_passwd.pl	2006-01-25 05:18:16.000000000 +0100
+++ MigrationTools-47-etc/migrate_passwd.pl	2007-11-05 17:17:05.897177898 +0100
@@ -37,7 +37,7 @@
 # Thanks to Peter Jacob Slot <peter%vision.auk.dk>.
 #
 
-require 'migrate_common.ph';
+require '%CONFDIR%migrate_common.ph';
 
 $PROGRAM = "migrate_passwd.pl";
 $NAMINGCONTEXT = &getsuffix($PROGRAM);
diff -ur MigrationTools-47/migrate_profile.pl MigrationTools-47-etc/migrate_profile.pl
--- MigrationTools-47/migrate_profile.pl	2006-01-25 05:18:16.000000000 +0100
+++ MigrationTools-47-etc/migrate_profile.pl	2007-11-05 17:17:05.897177898 +0100
@@ -36,7 +36,7 @@
 #
 #
 
-require 'migrate_common.ph';
+require '%CONFDIR%migrate_common.ph';
 
 $PROGRAM = "migrate_profile.pl";
 
diff -ur MigrationTools-47/migrate_protocols.pl MigrationTools-47-etc/migrate_protocols.pl
--- MigrationTools-47/migrate_protocols.pl	2006-01-25 05:18:16.000000000 +0100
+++ MigrationTools-47-etc/migrate_protocols.pl	2007-11-05 17:17:05.897177898 +0100
@@ -36,7 +36,7 @@
 #
 #
 
-require 'migrate_common.ph';
+require '%CONFDIR%migrate_common.ph';
 
 $PROGRAM = "migrate_protocols.pl";
 $NAMINGCONTEXT = &getsuffix($PROGRAM);
diff -ur MigrationTools-47/migrate_rpc.pl MigrationTools-47-etc/migrate_rpc.pl
--- MigrationTools-47/migrate_rpc.pl	2006-01-25 05:18:16.000000000 +0100
+++ MigrationTools-47-etc/migrate_rpc.pl	2007-11-05 17:17:05.900511003 +0100
@@ -36,7 +36,7 @@
 #
 #
 
-require 'migrate_common.ph';
+require '%CONFDIR%migrate_common.ph';
 
 $PROGRAM = "migrate_rpc.pl";
 $NAMINGCONTEXT = &getsuffix($PROGRAM);
diff -ur MigrationTools-47/migrate_services.pl MigrationTools-47-etc/migrate_services.pl
--- MigrationTools-47/migrate_services.pl	2006-01-25 05:18:16.000000000 +0100
+++ MigrationTools-47-etc/migrate_services.pl	2007-11-05 17:17:05.913843420 +0100
@@ -37,7 +37,7 @@
 #
 #
 
-require 'migrate_common.ph';
+require '%CONFDIR%migrate_common.ph';
 
 $PROGRAM = "migrate_services.pl";
 $NAMINGCONTEXT = &getsuffix($PROGRAM);
diff -ur MigrationTools-47/migrate_slapd_conf.pl MigrationTools-47-etc/migrate_slapd_conf.pl
--- MigrationTools-47/migrate_slapd_conf.pl	2006-01-25 05:18:16.000000000 +0100
+++ MigrationTools-47-etc/migrate_slapd_conf.pl	2007-11-05 17:17:05.923842731 +0100
@@ -36,7 +36,7 @@
 #
 #
 
-require 'migrate_common.ph';
+require '%CONFDIR%migrate_common.ph';
 
 $PROGRAM = "migrate_slapd_conf.pl";
 
diff -ur MigrationTools-47/README MigrationTools-47-etc/README
--- MigrationTools-47/README	2006-01-25 05:18:16.000000000 +0100
+++ MigrationTools-47-etc/README	2007-11-05 17:17:05.923842731 +0100
@@ -12,7 +12,7 @@
 
 You need perl to run these.
 
-Edit migrate_common.ph and change the following site-specific
+Edit @@CONFDIR%migrate_common.ph and change the following site-specific
 variables to reflect your installation:
 
 $DEFAULT_MAIL_DOMAIN 
